packages/headless/src/features/insight-search/insight-search-request.test.ts

```diff
@@ -397,4 +398,32 @@ describe('insight search request', () => {
       expect(params.context).toBe(expectedContext);
     });
   });
+
+  it('should enable #queryCorrection if did you mean is enabled and #queryCorrectionMode is `next`', async () => {
+    state.didYouMean.enableDidYouMean = true;
+    state.didYouMean.queryCorrectionMode = 'next';
+    expect(
+      (await buildInsightBaseRequest(state)).request.queryCorrection?.enabled
+    ).toBe(true);
+  });
+
+  it('should enable #automaticallyCorrect if did you mean is enabled and #queryCorrectionMode is `next` and #automaticallyCorrectQuery is true', async () => {
+    state.didYouMean.enableDidYouMean = true;
+    state.didYouMean.queryCorrectionMode = 'next';
+    state.didYouMean.automaticallyCorrectQuery = true;
+    expect(
+      (await buildInsightBaseRequest(state)).request.queryCorrection?.options
+        ?.automaticallyCorrect
+    ).toBe('whenNoResults');
+  });
+
+  it('should disable #automaticallyCorrect if did you mean is enabled and #queryCorrectionMode is `next` and #automaticallyCorrectQuery is false', async () => {
+    state.didYouMean.enableDidYouMean = true;
+    state.didYouMean.queryCorrectionMode = 'next';
```

Comment:
same remark, next is not being taking in consideration when setting `automaticallyCorrect ` in the request